kernel/mmanutils.c

```diff
@@ -63,34 +77,28 @@ int myst_teardown_mman(void)
 static ssize_t _map_file_onto_memory(
     int fd,
     off_t offset,
-    void* data,
-    size_t size)
+    void* addr,
+    size_t length,
+    int mmap_flags)
 {
-    ssize_t ret = -1;
+    ssize_t ret = 0;
     ssize_t bytes_read = 0;
```

Comment:
This should be some [checking](https://man7.org/linux/man-pages/man2/mmap.2.html) upfront:

> EACCES A file descriptor refers to a non-regular file.  Or a file
              mapping was requested, but fd is not open for reading.  Or
              MAP_SHARED was requested and PROT_WRITE is set, but fd is
              not open in read/write (O_RDWR) mode.  Or PROT_WRITE is
              set, but the file is append-only.

Reply:
Fixed. I added all the checks above.